src/compiler/program.ts

```diff
@@ -393,10 +393,17 @@ namespace ts {
         allDiagnostics?: Diagnostic[];
     }
 
-    export function isProgramUptoDate(program: Program | undefined, rootFileNames: string[], newOptions: CompilerOptions,
-        getSourceVersion: (path: Path) => string, fileExists: (fileName: string) => boolean, hasInvalidatedResolution: HasInvalidatedResolution): boolean {
-        // If we haven't create a program yet, then it is not up-to-date
-        if (!program) {
+    export function isProgramUptoDate(
+        program: Program | undefined,
+        rootFileNames: string[],
+        newOptions: CompilerOptions,
+        getSourceVersion: (path: Path) => string,
+        fileExists: (fileName: string) => boolean,
+        hasInvalidatedResolution: HasInvalidatedResolution,
+        hasChangedAutomaticTypeDirectiveNames: () => boolean,

```

Comment:
This is a completely public function, so it's a breaking change to add a new non-optional parameter.

Reply:
But this is the public api we are adding just now not present before this

Reply:
Oh, right. Must have been looking at a single commit.